manual_flagging/initial_flagging.py

```diff
@@ -121,153 +140,375 @@
 df = df.astype(conversion_dict)
 df["ptax_flag_original"].fillna(False, inplace=True)
 
-# Separate res and condo sales based on the indicator column
-df_res = df[df["indicator"] == "res"].reset_index(drop=True)
-df_condo = df[df["indicator"] == "condo"].reset_index(drop=True)
+# Filter to correct tris
+tri_stat_groups = {
+    tri: method
+    for tri, method in inputs["tri_stat_groups"].items()
+    if tri in inputs["run_tri"]
+}
 
-# Create condo stat groups. Condos are all collapsed into a single class,
-# since there are very few 297s or 399s
-condo_stat_groups = inputs["stat_groups"].copy()
-condo_stat_groups.remove("class")
+# Handle current methodology data manipulation if needed (temporary, need to find a better way)
+if "current" in tri_stat_groups.values():
+    # Calculate the building's age
+    current_year = datetime.datetime.now().year
+    df["bldg_age"] = current_year - df["yrblt"]
+
+    # Ingest new geographic groups
+    df_new_groups = pd.read_excel(
+        os.path.join(root, "QC_salesval_nbhds_round2.xlsx"),
+        usecols=["Town Nbhd", "Town Grp 1"],
+    ).rename(columns={"Town Nbhd": "nbhd", "Town Grp 1": "geography_split"})
+    df["nbhd"] = df["nbhd"].astype(int)
+    df = pd.merge(df, df_new_groups, on="nbhd", how="left")
+
+# - - - - - - -
+# Make correct filters and set up dictionary structure
+# - - - - - - -
+
+dfs_to_feature_creation = {}  # Dictionary to store DataFrames
+
+for tri, method in tri_stat_groups.items():
+    print(tri, method)
+
+    # Iterate over markets
+    for market in inputs["housing_run_type"]:
+        if method == "current":
+            key = f"df_tri{tri}_{market}_current"
+            # Filter by triad code and market type
+            triad_code_filter = df["triad_code"] == str(tri)
+            market_filter = df["class"].isin(inputs["housing_run_type_filters"][market])
+
+            dfs_to_feature_creation[key] = {
+                "df": df[triad_code_filter & market_filter],
+                "columns": inputs["stat_groups"]["current"][market]["columns"],
+                "iso_forest_cols": inputs["iso_forest"][
+                    "res" if "res" in market else "condos"
+                ],
+                "condos_boolean": market == "condos",
+            }
+
+        elif method == "og_mansueto":
+            # Collect all mansueto tris
+            mansueto_tris_to_flag = [
+                str(key)
+                for key, value in tri_stat_groups.items()
```

Comment:
I might be misunderstanding your point but I'll try to further explain the idea behind the code. The reason that we are collecting mansueto tris here:
```python
mansueto_tris_to_flag = [
    str(key)
    for key, value in tri_stat_groups.items()
    if value == "og_mansueto"
]
```
is that with the og_mansueto method, the groupings (rolling_window, class, township for res // rolling_window, township for condos) work no matter how many tris are included. This is is due to the fact that `og_mansueto` wasn't built to flexibly flag sales based on housing type, and that townships discretely make up a tri. So if we have tri 2 and tri 3 configured to `og_mansueto`, I thought to combine them into one data frame instead of iterating over each tri separately:

```python
 triad_code_filter = ~df["triad_code"].isin(mansueto_tris_to_flag)

df_res_og_mansueto = df[(df["indicator"] == "res") & triad_code_filter]
df_condo_og_mansueto = df[(df["indicator"] == "condo") & triad_code_filter]
```

Reply:
Gotcha! I think it's reasonable to want to handle the legacy method in a way that more closely reflects the structure of its data. In that case, I would advocate for splitting the `tri_stat_groups` loop by method, something like:

```python
# Split tris into groups according to their flagging methods
current_tris = [tri for tri, method in tri_stat_groups.items() if method == "current"]
og_mansueto_tris = [tri for tri, method in tri_stat_groups.items() if method == "og_mansueto"]

# Flag tris that use the most contemporary flagging method
for tri in current_tris:
    for market in inputs["housing_run_type"]:
        ...

# Flag tris that use the legacy method developed by the Mansueto Institute
triad_code_filter = ~df["triad_code"].isin(og_mansueto_tris)
...
```

That way we can avoid doing the confusing thing of iterating the `tri_stat_groups` object inside a loop that is already operating on its elements.

Reply:
Implemented this! I scrapped the loop for `og_mansueto` method.